Refresh the design of the pagination component

janfaracik opened the pull request:
**Before**
<img width="908" alt="image" src="https://user-images.githubusercontent.com/43062514/229311240-f8770b51-a699-42c0-9af9-f6f5adf7a1ed.png">

<img width="879" alt="image" src="https://user-images.githubusercontent.com/43062514/229311230-bd675016-f0bc-437a-90f2-7973bf9ed793.png">

**After**
<img width="893" alt="image" src="https://user-images.githubusercontent.com/43062514/229311210-307786fd-8e94-410d-90d8-2ac4911a35f2.png">

<img width="892" alt="image" src="https://user-images.githubusercontent.com/43062514/229311234-5bb6f70a-bcc6-4fe3-b888-6c73f5264458.png">

Small update to the pagination component to use similar stylings to Jenkins CI. Precursor to a larger blogs page update. 

* Uses .jenkins-button for the classes
* Now uses Ionicons
* Works better on mobile

MarkEWaite reviewed (approved):
This looks good to me.  I checked the navigation on the first page, second page, last page, and second to last page.  All behaved as expected on Chrome and on Firefox.

I was surprised on my mobile Chrome browser that the pagination numbers disappeared and were replaced with a single arrow left and a single arrow right.  That is different than the current pagination on mobile where the numbers are displayed across the bottom of the page in much the same way that they are displayed on my desktop browser.

Is that change intentional on the narrower screen that is on my Galaxy S22 cell phone in portrait mode?

NotMyFault commented:
![Screenshot 2023-04-02 at 16 25 56](https://user-images.githubusercontent.com/13383509/229359031-ec154ac0-a2b4-499a-bbc3-046e89d3b3af.png)
Right, this looks a bit odd.

Although it affects all phones, I guess, looks identical if I spin up an iPhone 13 Pro Max:

![Screenshot 2023-04-02 at 16 28 49](https://user-images.githubusercontent.com/13383509/229359191-5566ee92-3911-4c06-b624-e6a3c84769a3.png)

Maybe we can use the space between the navigation arrows and move pagination on mobile there?
![Screenshot 2023-04-02 at 16 30 11](https://user-images.githubusercontent.com/13383509/229359277-72dcc531-109e-4a7f-b4cd-4c3a83fa322a.png)

janfaracik commented:
Thanks for the reviews @MarkEWaite @NotMyFault! 

I removed the numbers from the pagination for mobile as in some situations the list can get very long/overflow, e.g. [if you're on page 50](https://user-images.githubusercontent.com/43062514/229364069-3c63b090-7862-4dd2-a0c9-3ef14a098606.png).

I've updated the mobile view to include a small page switcher control, which when tapped shows the entire list of pages and the user can then pick from any of them.

<img width="574" alt="image" src="https://user-images.githubusercontent.com/43062514/229365497-8d50630e-2402-4cb4-bdc1-2ef702cad908.png">

Let me know what you think, should be deployed in a few minutes.

NotMyFault reviewed (approved):
Thanks, that's much better. Especially if you don't need to click the next button a dozen of times to get through the pages:

https://user-images.githubusercontent.com/13383509/229367140-797aedb9-1c00-488a-8d42-ffa3abcb514c.mp4

MarkEWaite commented:
Confirmed on Firefox and Chrome with the same sequence tested as before.  Thanks!
